### PR TITLE
Order Model and Test Order Model files

### PR DESCRIPTION
**Pull request by shawn-mit**
# Description
This is the Order table as a  Model for Bangazon and the TestsCase for  Order Model
# Related Ticket(s)
List related ticket(s) that apply to the work done in this pull request
#2 
## Testing

[ x] There are new unit tests in this PR, and I verify that there is full coverage of all new code.
[ x] I certify that all existing tests pass

## Documentation

[ x] There is new documentation in this pull request that must be reviewed..

## Deploy Notes
manage.py  makemigrations
manage.py migrate 
mangage.py test bangazon_ui 

You may need to # comment out the URL.py file because routing is not finished.

```sh
git pull --prune
git checkout <feature_branch>
```

## Steps to Test
Outline the steps to test the code in this pull request.

 manage.py  makemigrations
manage.py migrate 
mangage.py test bangazon_ui 

You may need to # comment out the URL.py file because routing is not finished.

### CODE REVIEW

**Comment by ben-d-marks** [comment]
There are some issues. see my inline comments

File: bangazon/bangazon_ui/models/order_model.py [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+sys.path.append("../")
+from django.db import models
+from django.contrib.auth.models import User
+from bangazon_ui.models.customer_model import Customer
+from bangazon_ui.models.payment_type_model import PaymentType
+from bangazon_ui.models.product_model import Product
+
+
+
+class Order (models.Model):
```
Comment (Onodric) [inline bangazon/bangazon_ui/models/order_model.py]: `order` may be a keyword in python or django

File: bangazon/bangazon_ui/tests/test_order_model.py [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+
+
+class TestOrderModel(TestCase): 
+
+    """
+    This is a test to build implementation code for the Order Model. 
+
+    """
+
+def test_order(self):
```
Comment (ben-d-marks) [inline bangazon/bangazon_ui/tests/test_order_model.py]: This test won't run unless it's indented

File: bangazon/bangazon_ui/tests/test_order_model.py [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+
+    payment = PaymentType.objects.create(
+    customer=customer,
+    payment_type="Visa",
+    account_number="12345678",
+    ccv="111",
+    expiration_date="2017-01-01"
+    )
+
+    product = Product.objects.create(
```
Comment (ben-d-marks) [inline bangazon/bangazon_ui/tests/test_order_model.py]: Product does not take these arguments -- refer to the Product Model for specifics

**Review by ben-d-marks — COMMENTED** [review]
These i think will take care of the problem, as well as changes to the product model!

File: bangazon/bangazon_ui/tests/test_order_model.py (hunk 3 above)
Comment (ben-d-marks) [reply bangazon/bangazon_ui/tests/test_order_model.py]: Here are the relevant fields, with the suggested changes that let it work for me.
```
name = models.CharField(max_length=100, blank=True, default="")
price = models.DecimalField(max_digits=20, decimal_places=2)
description = models.TextField(max_length=300, default='')
quantity = models.IntegerField()
product_type =models.ForeignKey(ProductType, related_name="products", on_delete=models.CASCADE)
customer =models.ForeignKey(Customer, related_name="seller", on_delete=models.CASCADE)
```

File: bangazon/bangazon_ui/models/order_model.py [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+    """
+    The Order table pulls information from PaymentType and Product via a join 
+    """
+
+    customer = models.ForeignKey(User, on_delete=models.CASCADE)
```
Comment (ben-d-marks) [inline bangazon/bangazon_ui/models/order_model.py]: This needs to be an instance of Customer, which includes an instance of User.

Comment (juliakimchung) [reply bangazon/bangazon_ui/models/order_model.py]: Isn't the class name should be 'class Orders' instead of 'class Order'?

Comment (shawn-mit) [reply bangazon/bangazon_ui/models/order_model.py]: Thanks for that correction @Onodric  / @juliakimchung  checkout this doc for the Meta class line of code in the 'order_model.py' file . http://stackoverflow.com/questions/677172/verbose-name-plural-unexpected-in-a-model

**Comment by shawn-mit** [comment]
updated the urls.py as well -  CHECK THAT OUT !  grazie mille.

**Comment by danirenae** [comment]
🤖 🚀 good job!

**Comment by juliakimchung** [comment]
🚀 as you wish:) Thanks for the hard work!

**Pull request merged**